src/core/Vector.hpp

```diff
@@ -208,7 +214,85 @@ Vector<N, T> operator+(Vector<N, T> const &a, Vector<N, T> const &b) {
 }
 
 template <size_t N, typename T>
+void operator+=(Vector<N, T> &a, Vector<N, T> const &b) {
+  return detail::binary_op_assign(a, b, std::plus<T>());
+}
+
+template <size_t N, typename T>
 Vector<N, T> operator-(Vector<N, T> const &a, Vector<N, T> const &b) {
   return detail::binary_op(a, b, std::minus<T>());
 }
+
+template <size_t N, typename T> Vector<N, T> operator-(Vector<N, T> const &a) {
+  Vector<N, T> ret;
+
+  std::transform(a.begin(), a.end(), ret.begin(),
+                 [](T const &v) { return -v; });
+
+  return ret;
+}
+
+template <size_t N, typename T>
+void operator-=(Vector<N, T> &a, Vector<N, T> const &b) {
+  return detail::binary_op_assign(a, b, std::minus<T>());
+}
+
+/* Scalar multiplication */
+template <size_t N, typename T>
+Vector<N, T> operator*(T const &a, Vector<N, T> const &b) {
+  Vector<N, T> ret;
+
+  std::transform(b.begin(), b.end(), ret.begin(),
+                 [a](T const &val) { return a * val; });
+
+  return ret;
+}
+
+template <size_t N, typename T>
+Vector<N, T> operator*(Vector<N, T> const &b, T const &a) {
+  Vector<N, T> ret;
+
+  std::transform(b.begin(), b.end(), ret.begin(),
+                 [a](T const &val) { return a * val; });
+
+  return ret;
+}
+
+template <size_t N, typename T> void operator*=(Vector<N, T> &b, T const &a) {
```

Comment:
The assignment operators should return a reference to the result, see http://en.cppreference.com/w/cpp/language/operator_assignment